### ISSUE TITLE
Add SigV4 for SES SMTP passwords

### ISSUE THREAD

**Issue by n3ph** [terraform-providers/terraform-provider-aws#11143]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

As SigV2 is deprecated and replaced by [SigV4](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/smtp-credentials.html#smtp-credentials-convert) the [used signature algorithm](https://github.com/terraform-providers/terraform-provider-aws/blob/master/aws/resource_aws_iam_access_key.go#L200-L216) should be updated. Since the new algorithm intends to sign a password for a specific AWS region it should be possible to get signed SES SMTP passwords per defined region.

### affected Resource

* aws_iam_access_key

### Potential Terraform Configuration

```hcl
resource "aws_iam_user" "test" {
  name = "test"
  path = "/test/"
}

resource "aws_iam_access_key" "test" {
  user             = aws_iam_user.test.name
  ses_smtp_regions = ["eu-central-1", "us-east-1"]
}

output "aws_iam_smtp_password_eu_central_1" {
  value = [for ses_smtp_password in aws_iam_access_key.test.ses_smtp_passwords :
  ses_smtp_password.secret if ses_smtp_password.region == "eu-central-1"][0]
}
```

### References

* Relates to #10903
* Relates to #11930
* Closed by #11144

**Comment by n3ph** [comment]
As suggested in https://github.com/terraform-providers/terraform-provider-aws/pull/11144#pullrequestreview-345744653 I simplified the solution to use the current provider region as a parameter for the SigV4 algorithm.
Code base is heavily simplified and users aren't going to be confused by the multiple region capability of that resource.

### Potential Terraform Configuration

```hcl
resource "aws_iam_user" "test" {
  name = "test"
  path = "/test/"
}

resource "aws_iam_access_key" "test" {
  user             = aws_iam_user.test.name
}

output "aws_iam_smtp_password_v4" {
  value = aws_iam_access_key.test.ses_smtp_password_v4
}
```

**Comment by bflad** [comment]
Support for generating SES SMTP v4 passwords into a new `ses_smtp_password_v4` attribute has been merged and will release with version 2.50.0 of the Terraform AWS Provider. Thanks to @n3ph for the implementation. 👍 

Please note: The password value is region-specific to the provider region of the `aws_iam_access_key` resource. To work across multiple regions, separate resources are necessary.
